lib/helpers.js

```diff
@@ -102,12 +102,17 @@ export async function createCapabilities({profileId, request}) {
 //    and one for its profile agent (using its `profileAgentContent`)
 // 4. creates a new credentials EDV
 // 5. stores capabilities to access capabilities EDV in the profileAgent
-export const createProfile = async ({profileAgentContent, profileContent}) => {
+export const createProfile = async (
+  {profileAgentContent, profileContent, pManager}) => {
   const t0 = Date.now();
-
-  const {id: profileId, meters} = await profileManager.createProfile(
+  let manager;
+  if(pManager) {
+    manager = pManager;
+  } else {
+    manager = profileManager;
+  }
```

Comment:
```suggestion
  const manager = pManager || profileManager;
```

Reply:
Actually, ignore that suggestion.
I don't think this is quite the right approach. I get what it's trying to do - either use the global `profileManager` from `state.js`, or be able to pass one in.
But I think you should refactor this not here in the function, but in the _callers_ of the function.
Meaning, `createProfile` should have just one input parameter, `profileManager`. And then the callers can decide whether to pass in the profileManager from `state.js`, or their own (like the CLI does).

Reply:
I would suggest looking into refactoring whatever is using this library that has its "own profile manager", if possible -- not making any of these changes here.